src/redux/missionSlice.js

```diff
@@ -0,0 +1,93 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import HttpService from '../api/service-handler';
+
+export const initialState = {
+  missions: [],
+  status: 'idle',
+  error: null,
+};
+
+export const fetchMissions = createAsyncThunk(
+  'missions/fetchMissions',
+  async () => {
+    const response = await HttpService.getMissions('missions');
+    return response;
+  },
+);
+
+const missionSlice = createSlice({
+  name: 'missions',
+  initialState,
+  reducers: {
+    setMissions: (state, action) => {
+      const mission = {
+        id: action.payload.mission_id,
+        missionName: action.payload.mission_name,
+        description: action.payload.description,
+        url: action.payload.wikipedia,
+        reserved: false,
+      };
```

Comment:
- [OPTIONAL] I would encourage you to take full advantage of destructuring here, so your code will satisfy the KISS. It would look like:
```
  setMissions: (state, {payload}) => {
      const {mission_id, mission_name, description, wikipedia, } = payload
      const mission = {
        id: mission_id,
        missionName: mission_name,
        description: description,
        url: wikipedia,
        reserved: false,
      };
```